e-commerce_fake-store/src/pages/home-page/index.js

```diff
@@ -0,0 +1,67 @@
+import React, { useEffect, useCallback } from "react";
+import Product from "../../components/product-card";
+import Navbar from "../../components/navbar";
+// import {useHistory } from "react-router-dom";
+
+const Home = () => {
+  const [allProducts, setAllProducts] = React.useState([]);
+  const [search, setSearch] = React.useState("");
+  // const history = useHistory();
+
+  // api varibale
+  const api = "https://fakestoreapi.com/";
+
+  const getAllProducts = useCallback(async () => {
+    const response = await fetch(`${api}` + "products");
+    const data = await response.json();
+    setAllProducts(data);
+  }, []);
+
+  useEffect(() => {
+    getAllProducts();
+  }, [getAllProducts]);
+  return (
+    <>
+      <Navbar
+        variant={"loggedin"}
+        cartcount={localStorage.getItem("cartcount")}
+      />
+      <div>
+        <label>Search</label>
+        <input
+          type="text"
+          placeholder="search..."
+          className="inputfields"
+          onChange={(e) => setSearch(e.target.value)}
+        ></input>
+      </div>
+      <h1>
+        Welcome {JSON.parse(localStorage.getItem("alluserdata")).fullname}
+      </h1>
+      {allProducts
+        .filter((product) => {
+          if (search === "") {
```

Comment:
You are returning the same '**product**' in both the if and else blocks. Nothing different. So, you can squeeze the condition check in that case like this:

```
if (!search || product.title.toLowerCase().includes(search.toLowerCase())) {
    return product;
} 
```

Reply:
Modified.